### ISSUE TITLE
HTML minification breaks SVG data: URLs due to XML parse error

### ISSUE THREAD

**Issue by westonruter** [W3EDGE/w3-total-cache#162]
I believe the HTML Minification is trying to remove useless self-closing slashes form HTML tags. However, when SVG XML is embedded inside of a `data:` URL, this actually breaks the page.

Consider this source HTML:

```html
<amp-img src="https://mauteam.org/wp-content/uploads/2019/08/mauteam-logo-75-V.png" alt="" width="75" height="75" class="amp-wp-enforced-sizes i-amphtml-layout-intrinsic i-amphtml-layout-size-defined" layout="intrinsic" i-amphtml-layout="intrinsic">
	<i-amphtml-sizer class="i-amphtml-sizer">
		<img alt="" aria-hidden="true" class="i-amphtml-intrinsic-sizer" role="presentation" src="data:image/svg+xml;charset=utf-8,<svg height=&quot;75&quot; width=&quot;75&quot; xmlns=&quot;http://www.w3.org/2000/svg&quot; version=&quot;1.1&quot;/>">
	</i-amphtml-sizer>
	<noscript><img src="https://mauteam.org/wp-content/uploads/2019/08/mauteam-logo-75-V.png" alt width="75" height="75" class=""></noscript>
</amp-img>
```

And the same HTML minified by W3 Total Cache:

```html
<amp-img src=https://mauteam.org/wp-content/uploads/2019/08/mauteam-logo-75-V.png alt width=75 height=75 class="amp-wp-enforced-sizes i-amphtml-layout-intrinsic i-amphtml-layout-size-defined" layout=intrinsic i-amphtml-layout=intrinsic>
	<i-amphtml-sizer class=i-amphtml-sizer>
		<img alt aria-hidden=true class=i-amphtml-intrinsic-sizer role=presentation src="data:image/svg+xml;charset=utf-8,<svg height=&quot;75&quot; width=&quot;75&quot; xmlns=&quot;http://www.w3.org/2000/svg&quot; version=&quot;1.1&quot;>">
	</i-amphtml-sizer>
	<noscript><img src=https://mauteam.org/wp-content/uploads/2019/08/mauteam-logo-75-V.png alt width=75 height=75 class></noscript>
</amp-img>
```

The key difference is in the the sizer `img` which has an SVG `data:` URL:

```diff
- version=&quot;1.1&quot;/>">
+ version=&quot;1.1&quot;>">
```

The lack of the trailing slash causes an XML parse error, and the image fails to display.

What's more is that this causes an AMP validation error, as the validator spec requires the `src` to follow this [pattern](https://github.com/ampproject/amphtml/blob/1a5aaaf43abd3992ce58cb4a1ec716e399f9a15c/validator/validator-main.protoascii#L4879):

```
  attrs: {
    name: "src"
    value_regex: "data:image\\/svg\\+xml;charset=utf-8,<svg height=\"\\d+\" width=\"\\d+\" xmlns=\"http:\\/\\/www\\.w3\\.org\\/2000\\/svg\" version=\"1\\.1\"\\/>|data:image\\/svg\\+xml;charset=utf-8,<svg height='100' width='300' xmlns='http:\\/\\/www\\.w3\\.org\\/2000\\/svg' version='1\\.1'\\/>"
    mandatory: true
  }
```

This came up on the AMP plugin support forum: https://wordpress.org/support/topic/error-images-4/

And on the AMP plugin GitHub repo: https://github.com/ampproject/amp-wp/issues/4491

**Comment by westonruter** [comment]
What's the ETA on this fix? We're getting reports about it daily.

**Comment by megaz555** [comment]
Yes, please fix this as soon as possible.

**Comment by bwmarkle** [comment]
Hi @westonruter and everyone,

> What's the ETA on this fix? We're getting reports about it daily.

Sorry about the delay in getting back. This should be fixed in our next patch release, which will be either this week or next. I touched base with the developer working on this one, and they mentioned they're having a little trouble with the _negative lookahead_ - it sounds like the fix is taking a little longer because of the tricky regex.

Thanks again for following up! This ticket is on our radar and we're on it.

**Comment by cssjoe** [comment]
Thank you @westonruter for reporting this issue.  A fix has been included in version 0.13.3, which is being released now.  Please let us know if you see any other issues.  We will be happy to address them.
